flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/JobSubmitTest.java

```diff
@@ -170,4 +170,43 @@ public void initializeOnMaster(ClassLoader loader) throws Exception {
 			fail(e.getMessage());
 		}
 	}
+
+	/**
+	 * Verifies failure when the client and job manager versions mismatch
+	 */
+	@Test
+	public void testFailureClientJobManagerVersionMismatch() {
+		try {
+			// create a simple job graph
+
+			JobVertex jobVertex = new JobVertex("Vertex that fails in initializeOnMaster") {
+
+				@Override
+				public void initializeOnMaster(ClassLoader loader) throws Exception {
+					throw new RuntimeException("test exception");
+				}
+			};
+
+			jobVertex.setInvokableClass(Tasks.NoOpInvokable.class);
+			JobGraph jg = new JobGraph("test job", jobVertex);
+
+			// submit the job
+			Future<Object> submitFuture = jobManager.ask(new JobManagerMessages.SubmitJob(jg, false, "RANDOM_CLIENT_VERSION"), timeout);
+			try {
+				Await.result(submitFuture, timeout);
+			}
+			catch (JobExecutionException e) {
+				// that is what we expect
+				// test that the exception nesting is not too deep
+				assertTrue(e.getCause() == null);
+			}
+			catch (Exception e) {
+				fail("Wrong exception type");
+			}
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+			fail(e.getMessage());
```

Comment:
Why do you print the stack trace and let the test fail with an `AssertionError` instead of letting the test fail with the true reason, e.g. `e`.